### ISSUE TITLE
[Post-SC enhancement] add testing capability to JITted functions

### ISSUE THREAD

**Issue by DavidPoliakoff** [hfinkel/llvm-project-cxxjit#4]
One thing we're winning is that we drive down development compile times in situations where we specialize a function a lot of ways. However, we lose that if it we push compile-time errors down to runtime. It would be interesting if we could provide a candidate set of JIT parameters, and at compile time just make sure that such a set of parameters doesn't cause a compiler error. So if we have

```
template<int N>
[[clang::jit]] void matmul(...)
{
}
```

If we had some way to tell the compiler "hey, make sure that if N is 2, we don't have a compiler error," we'd have a nice user experience, the user just pays the cost of trying to instantiate one template, but still gets some help from the compiler.
